### ISSUE TITLE
Unable to delete some seemingly empty folders

### ISSUE THREAD

**Issue by rsgoncalves** [metadatacenter/cedar-project#347]
This occurred with a bunch of folders during some test runs. I wasn't able to reproduce precisely unfortunately. In one case, I deleted a template contained in a folder, and tried to delete the folder right after, but got the "folder non-empty" warning. Have a look at [this screencast](https://youtu.be/v54-RAQNnJ4).

**Comment by egyedia** [comment]
If you can reproduce it, let me know, and we will run some queries to see what is happening there.
It can be related to a visibility issue that we talked over with Marcos and Martin, and postponed for now.

**Comment by rsgoncalves** [comment]
Managed to reproduce it today with @willrett. Here are the steps:

* Login as User 1
* Create a folder `foo` and share (with write permissions) it with User 2
* Login as User 2
* Navigate to `foo` and create a template inside that folder
* Logout and login as User 1
* Delete `foo`

The error message says the folder is not empty, but you can see it's empty on User 1's side. The message should indicate that the folder has been shared and contains resources in some other user's folder. But since the folder was shared, I wonder if the owner should have permissions to resources created within it?

**Comment by egyedia** [comment]
Yes, than this is the same visibility issue that we saw with Marcos, and discussed with Martin.
Currently owning a folder does not mean you can access its contents.
It is not exactly the same, but on a Unix filesystem if the permission flags are set to 077, the owner can not do a thing with that resource, even though he/she is the owner.
We will probably need to review our permission rules, and fix this. It is in the medium term plans - at least it is in my plans.

**Comment by rsgoncalves** [comment]
In Google Drive, folder permissions are inherited, as in, if User 1 shares a folder with User 2, and User 2 creates resources within that folder,* those resources have the same visibility as the folder itself (so User 1 can see them).

* When User 2 attempts to create a resource in the shared folder, a warning comes up stating "The created item will have the same sharing permissions as the selected folder."

**Comment by graybeal** [comment]
While I don't love Google permissions in this case, I think we have to consider following their pattern, just because we follow many of their other UX patterns. And it is less technical to explain than the Unix pattern.

On Mar 9, 2017, at 10:58 AM, Rafael Gonçalves <notifications@github.com<mailto:notifications@github.com>> wrote:


In Google Drive, folder permissions are inherited, as in, if User 1 shares a folder with User 2, and User 2 creates resources within that folder,* those resources have the same visibility as the folder itself (so User 1 can see them).

  *   When User 2 attempts to create a resource in the shared folder, a warning comes up stating "The created item will have the same sharing permissions as the selected folder."

**Comment by martinjoconnor** [comment]
Morphing this into an issue that addresses rethinking folder organization and permissions and moving to Backlog because it is unlikely to be doable in 1.1 timeframe.

**Comment by willrett** [comment]
I agree that the permissions should be inherited.  If you are given access to a folder, you should have the same access to anything inside that folder.  

I see this often because the Protractor tests are creating, sharing and deleting resources and folders.   Anything not cleaned up by the tests is really hard to delete by hand.  You get the message that your folder is not empty and cannot be deleted, but it looks empty to you.  Some other user  created something that I cannot see and left it there.
